cusub_cortex/state_machine/src/tasks/jiangshi_det.py

```diff
@@ -179,9 +195,78 @@ def execute(self, userdata):
         return "backed_up"
 
 
-# TODO
-class Revisit(Objective):
+class Retrace(Objective):
```

Comment:
Is each task going to have its own Retrace obj? Would it make sense to have a super class to avoid repeating code?

Reply:
We should talk about this. I'm currently duplicating these things: 

- Each task needs a timeout object to enter Retrace (usually in the Approach Objective). 

- Each task will have a Retrace function. What would be best is to inherit a general retrace function. We'd need to add some complexity to it, just because for some tasks we only expect one dobject (Jiangshi), others we except multiple dobjects, but only one of each type (Droppers), and others where we expect multiple of a certain dobject (Start Gate).